compose/service.py

```diff
@@ -988,6 +984,38 @@ def parse_volume_spec(volume_config):
 
     return VolumeSpec(external, internal, mode)
 
+
+def build_volume_from(volume_from_spec):
+    """
+    volume_from can be either a service or a container. We want to return the
+    container.id and format it into a string complete with the mode.
+    """
+    if isinstance(volume_from_spec.source, Service):
+        containers = volume_from_spec.source.containers(stopped=True)
+        if not containers:
+            return ["{}:{}".format(volume_from_spec.source.create_container().id, volume_from_spec.mode)]
+
+        container = containers[0]
+        return ["{}:{}".format(container.id, volume_from_spec.mode)]
+    elif isinstance(volume_from_spec.source, Container):
+        return ["{}:{}".format(volume_from_spec.source.id, volume_from_spec.mode)]
```

Comment:
Yep, I removed it when I was changing them to return statements, because I realised, I _think_ that it isn't a valid case/return value.

By this point in the code, the `volume_from`  *has* to be either a service or container as done by this bit of code, https://github.com/docker/compose/blob/master/compose/project.py#L176 which gets called before it's value is passed into `Service`. So there should not be a situation where we `return []`, does that make sense?